examples/bits.rs

```diff
@@ -1,98 +1,207 @@
+        println!("Character {}: {}", char_num, codepoint);
+        // Each character has a variable width, so we need to find that.
+        let byte_count = codepoint.len_utf8();
+        // And then collect the bytes of the string that make up the character.
+        // `start` gives us the starting position in the text sequence, and
+        // `byte_count` gives us the length in bytes of the character, so we
+        // need to select the range beginning at `start`, running for
+        // `byte_count`. Another style of writing this that you might see in
+        // Rust libraries is `[start ..][.. length]`.
+        let row: &[u8] = &TEXT.as_bytes()[start .. start + byte_count];
+
+        // For each byte (`u8`), we use `bitvec` to make a view into its bits.
+        // `bitvec` provides the `.bits::<_>()` method on Rust integers for easy
+        // access to its view types.
+        //
+        // The `LittleEndian` means that the view moves from least significant
+        // bit to most significant. Since we want to display on screen the most
+        // significant bit on the left, and the least on the right, the turtle
+        // will have to move from right to left to match.
+        //
+        // These types describe different ways to view the same data. You can
+        // read more about them in the `bitvec` docs, and at Wikipedia:
+        // https://docs.rs/bitvec/0.16.1/bitvec/cursor/index.html
+        // https://en.wikipedia.org/wiki/Endianness#Bit_endianness
+        for byte in row {
+            println!("  Byte {:02}:\n    Value: 0x{:02X}\n    Bits: {:08b}", row_num, byte, byte);
+
+            let bits: &BitSlice<_, _> = byte.bits::<LittleEndian>();
 
-impl BitVec {
-    /// Create a new BitVec with the given size in bits
-    /// Note that the actual size of the vector is in bytes so `bits` will be rounded up to the
-    /// nearest multiple of 8 >= bits.
-    pub fn with_capacity(bits: usize) -> Self {
-        if bits > 0 {
-            // Ceiling division of bits / BITS so that we get >= bits capacity
-            // From: https://stackoverflow.com/a/2745086/551904
-            let capacity = 1 + ((bits - 1) / BITS);
-            Self { bytes: vec![0; capacity] }
-        } else {
-            Default::default()
+            // Then we draw the byte's bits as a row
+            draw_row(&mut turtle, bits);
+            // And reset the turtle to the right edge.
+            turtle.set_x(right_edge);
+
+            row_num += 1;
         }
     }
 
-    /// Returns the size in bits of the array
-    pub fn len(&self) -> usize {
-        self.len_bytes() * BITS
-    }
+    // `bitvec` can look at more than just `u8`. Let's try looking at the bits
+    // that represent a number!
+    //
+    // But `bitvec` doesn't know how to view all numbers. The standard library
+    // provides a function `to_bits(f64) -> u64`, which turns it into a number
+    // `bitvec` does know how to view.
```

Comment:
What do you think of changing this to the following:

```markdown
Some numbers like f64 have a special representation in bits, so let's
extract that using the `to_bits(f64) -> u64` method on `f64` provided
by the Rust standard library. This method turns an f64 value into a
number that we can view with `bitvec`.
```

P.S. I think it was a great choice to use `f64` here. I don't think a lot of people know about `to_bits` and I'm glad we can teach it to them! :smile:

Reply:
How about

> Some numbers, like `f64`, have special rules for their representation in bits. Since `bitvec` does not know those rules, it does not work on `f64`. However, Rust provides a method `to_bits(f64) -> u64` that lets us forget those rules and use the number's storage as ordinary bits.
>
> You can read more about the rules for `f64`'s storage in memory, and behavior in programs, here: (link)

I think that using "special rules" over "special representation" will convey that it's ordinary memory, just interpreted differently, and that we are just removing that lens in order to look at it as bare data rather than a meaningful number.

Reply:
Great revision! What do you think of the following?

```markdown
Some numbers, like `f64`, have special rules for their representation in
bits. The `bitvec` crate only cares about the raw bits, not how `f64` uses
them, so it does not operate on `f64` directly. To get the raw bits from
`f64`, Rust provides a method `to_bits(f64) -> u64` that lets us forget
those rules and use the number's storage as ordinary bits.

You can read more about the rules for `f64`'s storage in memory,
and behavior in programs, here: (link)
```